src/entities/friendship_history.rs

```diff
@@ -23,30 +24,40 @@ pub struct FriendshipHistory {
     pub metadata: Option<Json<HashMap<String, String>>>,
 }
 
-#[cfg_attr(test, faux::methods)]
 impl FriendshipHistoryRepository {
     pub fn new(db: Arc<Option<DBConnection>>) -> Self {
         Self { db_connection: db }
     }
 
-    pub async fn create(
+    pub fn create_query<'a>(
         &self,
         friendship_id: Uuid,
-        event: &str,
-        acting_user: &str,
+        event: &'a str,
+        acting_user: &'a str,
         metadata: Option<Json<HashMap<String, String>>>,
-    ) -> Result<(), sqlx::Error> {
-        let db_conn = DatabaseComponent::get_connection(&self.db_connection);
-        match sqlx::query(
+    ) -> Query<'a, Postgres, sqlx::postgres::PgArguments> {
+        sqlx::query(
                 "INSERT INTO friendship_history (id,friendship_id, event, acting_user, metadata) VALUES ($1,$2,$3,$4,$5)",
         )
         .bind(Uuid::parse_str(generate_uuid_v4().as_str()).unwrap())
         .bind(friendship_id)
         .bind(event)
         .bind(acting_user)
         .bind(metadata)
-        .execute(db_conn)
-        .await
+    }
+
+    pub async fn create(
+        &self,
+        friendship_id: Uuid,
+        event: &str,
+        acting_user: &str,
+        metadata: Option<Json<HashMap<String, String>>>,
```

Comment:
Let's add an alias for the `Json<...>` metadata type.

Reply:
Revamping this comment above. ☝🏻

Reply:
Sorry, had missed this comment, I'll create it now